xds/internal/balancer/edsbalancer/configbuilder_test.go

```diff
@@ -0,0 +1,164 @@
+/*
+ *
+ * Copyright 2021 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package edsbalancer
+
+import (
+	"fmt"
+	"testing"
+
+	"google.golang.org/grpc/balancer"
+	"google.golang.org/grpc/balancer/roundrobin"
+	"google.golang.org/grpc/xds/internal"
+	"google.golang.org/grpc/xds/internal/balancer/priority"
+	xdsclient "google.golang.org/grpc/xds/internal/client"
+
+	_ "google.golang.org/grpc/xds/internal/balancer/clustermanager" // Register the xds_cluster_manager balancer
+	_ "google.golang.org/grpc/xds/internal/balancer/weightedtarget" // Register the weighted_target balancer
+)
+
+const (
+	localityCount      = 4
+	addressPerLocality = 2
+)
+
+var (
+	testLocalityIDs []internal.LocalityID
+	testAddressStrs [][]string
+	testEndpoints   [][]xdsclient.Endpoint
+
+	testLocalitiesP0, testLocalitiesP1 []xdsclient.Locality
+)
+
+func init() {
+	for i := 0; i < localityCount; i++ {
+		testLocalityIDs = append(testLocalityIDs, internal.LocalityID{Zone: fmt.Sprintf("test-zone-%d", i)})
+		var (
+			addrs []string
+			ends  []xdsclient.Endpoint
+		)
+		for j := 0; j < addressPerLocality; j++ {
+			addr := fmt.Sprintf("addr-%d-%d", i, j)
+			addrs = append(addrs, addr)
+			ends = append(ends, xdsclient.Endpoint{
+				Address:      addr,
+				HealthStatus: xdsclient.EndpointHealthStatusHealthy,
+			})
+		}
+		testAddressStrs = append(testAddressStrs, addrs)
+		testEndpoints = append(testEndpoints, ends)
+	}
+
+	testLocalitiesP0 = []xdsclient.Locality{
+		{
+			Endpoints: testEndpoints[0],
+			ID:        testLocalityIDs[0],
+			Weight:    20,
+			Priority:  0,
+		},
+		{
+			Endpoints: testEndpoints[1],
+			ID:        testLocalityIDs[1],
+			Weight:    80,
+			Priority:  0,
+		},
+	}
+	testLocalitiesP1 = []xdsclient.Locality{
+		{
+			Endpoints: testEndpoints[2],
+			ID:        testLocalityIDs[2],
+			Weight:    20,
+			Priority:  1,
+		},
+		{
+			Endpoints: testEndpoints[3],
+			ID:        testLocalityIDs[3],
+			Weight:    80,
+			Priority:  1,
+		},
```

Comment:
I don't see any of this being used in this test. Am I missing something?

Reply:
Deleted, these 2 and `testAddressStrs`.